projects/src/transmitter.cc

```diff
@@ -0,0 +1,157 @@
+#include "transmitter.h"
+#include "hal.h"
+#include "packet/serialize.h"
+#include "usbconfig.h"
+
+// This define can be useful when sizing message mailbox and memory pools
+#define ASSERT_MESSAGE_MEMORY_LIMIT FALSE
+
+namespace message {
+Transmitter::Transmitter() :
+m_thread(nullptr),
+m_bytes_written(0) {
+    initialize();
+
+    // Initialize a serial-over-USB CDC driver.
+    sduObjectInit(&SDU1);
+    sduStart(&SDU1, &serusbcfg);
+
+    // Activate the USB driver and then the USB bus pull-up on D+.
+    // Note, a delay is inserted in order to not have to disconnect the cable
+    // after a reset.
+    board_usb_lld_disconnect_bus();   //usbDisconnectBus(serusbcfg.usbp);
+    chThdSleepMilliseconds(1500);
+    usbStart(serusbcfg.usbp, &usbcfg);
+    board_usb_lld_connect_bus();      //usbConnectBus(serusbcfg.usbp);
+
+    // Block USB device until ready
+    while ((SDU1.config->usbp->state != USB_ACTIVE) || (SDU1.state != SDU_READY)) {
```

Comment:
I find that negating the desired state makes the condition easier to read. 
`while (!(SDU1.config->usbp->state == USB_ACTIVE && SDU1.state == SDU_READY)) {`